hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/utilops/SysFileOverrideOp.java

```diff
@@ -0,0 +1,143 @@
+import static com.hedera.services.bdd.suites.HapiSuite.GENESIS;
+import static com.hedera.services.bdd.suites.HapiSuite.ONE_HBAR;
+import static com.hedera.services.bdd.suites.utils.sysfiles.serdes.StandardSerdes.SYS_FILE_SERDES;
+import static java.util.Objects.requireNonNull;
+
+import com.google.protobuf.ByteString;
+import com.hedera.services.bdd.spec.HapiSpec;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.OptionalLong;
+import java.util.function.Supplier;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * A utility operation to override the contents of a dynamic system file, remembering the initial contents for
+ * later restoration.
+ */
+public class SysFileOverrideOp extends UtilOp {
+    private static final Logger log = LogManager.getLogger(SysFileOverrideOp.class);
+
+    public enum Target {
+        FEES {
+            @Override
+            public long number() {
+                return 111L;
+            }
+        },
+        EXCHANGE_RATES {
+            @Override
+            public long number() {
+                return 112L;
+            }
+        },
+        THROTTLES {
+            @Override
+            public long number() {
+                return 123L;
+            }
+        },
+        PERMISSIONS {
+            @Override
+            public long number() {
+                return 122L;
+            }
+        };
+
+        public abstract long number();
+    }
+
+    private final Target target;
+    private final Supplier<String> overrideSupplier;
+
+    private boolean autoRestoring = true;
+    private byte[] originalContents;
+
+    /**
+     * Creates a new operation to override the contents of the given system file on the target network of the given,
+     * but <b>not</b> automatically restore the original contents when the spec that submitted this operation completes.
+     * @param target the system file to override
+     * @param overrideSupplier the supplier of the new contents of the system file
+     * @return the new operation
+     */
+    public static SysFileOverrideOp withoutAutoRestoring(
+            @NonNull final Target target, @NonNull final Supplier<String> overrideSupplier) {
+        final var op = new SysFileOverrideOp(target, overrideSupplier);
+        op.autoRestoring = false;
+        return op;
+    }
+
+    public SysFileOverrideOp(@NonNull final Target target, @NonNull final Supplier<String> overrideSupplier) {
+        this.target = requireNonNull(target);
+        this.overrideSupplier = requireNonNull(overrideSupplier);
+    }
+
+    @Override
+    protected boolean submitOp(@NonNull final HapiSpec spec) throws Throwable {
+        allRunFor(spec, getFileContents("0.0." + target.number()).consumedBy(bytes -> this.originalContents = bytes));
+        log.info("Took snapshot of {}", target);
+        final var styledContents = overrideSupplier.get();
+        if (styledContents != null) {
```

Comment:
Would a log be desirable for either `styledContents == null` (or maybe `isBlank` too)?

(Is there a use case for the supplied new contents to be empty or blank?)